### ISSUE TITLE
Send Test Email from SMTP Setup Error - The mail system returned the following error: "A call to MailKit.Net.Smtp.SmtpClient.Connect failed with this message: This is usually a temporary error during hostname resolution and means that the local server did not receive a response from an authoritative server".

### ISSUE THREAD

**Issue by DavidRoys** [microsoft/navcontainerhelper#1348]
**Describe the issue**
Trying to send a test email from a recently created BC Container returns the error message: 

The mail system returned the following error: "A call to MailKit.Net.Smtp.SmtpClient.Connect failed with this message: This is usually a temporary error during hostname resolution and means that the local server did not receive a response from an authoritative server".

**Scripts used to create container and cause the issue**
```
$containerName = 'bcau'
$User = "davidr"
$PWord = ConvertTo-SecureString -String "Password1" -AsPlainText -Force
$Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $User, $PWord
$auth = 'UserPassword'
$artifactUrl = Get-BcArtifactUrl -type 'Sandbox' -version '' -country 'au' -select 'SecondToLastMajor'
$licenseFile = 'https://www.dropbox.com/s/f5xuxd80lq82n3k/BC%20Dev%20License%2026-SEP-2020.flf?dl=1'

New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -credential $credential `
    -auth $auth `
    -artifactUrl $artifactUrl `
    -imageName 'bcnzimage' `
    -includeTestToolkit `
    -assignPremiumPlan `
    -licenseFile $licenseFile `
    -updateHosts `
    -additionalParameters @("--restart always")
```

**Full output of scripts**
I don't have the output of the script but I ran the Get-BCContainerDebugInfo and it gave the following:
```
{
    "container.logs":  [
                           "Initializing...",
                           "Setting host.docker.internal to 192.168.1.78 in container hosts file (copy from host hosts file)",
                           "Setting gateway.docker.internal to 192.168.1.78 in container hosts file (copy from host hosts file)",
                           "Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)",
                           "Setting host.containerhelper.internal to 172.25.192.1 in container hosts file",
                           "Starting Container",
                           "Hostname is bcau",
                           "PublicDnsName is bcau",
                           "Using NavUserPassword Authentication",
                           "Starting Local SQL Server",
                           "Starting Internet Information Server",
                           "Creating Self Signed Certificate",
                           "Self Signed Certificate Thumbprint 5D83150BEB818B746BA127F1867197D1CF084D23",
                           "Modifying Service Tier Config File with Instance Specific Settings",
                           "Starting Service Tier",
                           "Registering event sources",
                           "Creating DotNetCore Web Server Instance",
                           "Enabling Financials User Experience",
                           "Enabling rewrite rule: Don\u0027t rewrite system files",
                           "Enabling rewrite rule: Already have tenant specified",
                           "Enabling rewrite rule: Hostname (without port) to tenant",
                           "Using license file \u0027c:\\run\\my\\license.flf\u0027",
                           "Import License",
                           "Dismounting Tenant",
                           "Mounting Tenant",
                           "Mounting Database for default on server localhost\\SQLEXPRESS",
                           "Sync\u0027ing Tenant",
                           "Tenant is Operational",
                           "Creating http download site",
                           "Setting SA Password and enabling SA",
                           "Creating davidr as SQL User and add to sysadmin",
                           "Creating SUPER user",
                           "Assign Premium plan for DAVIDR",
                           "Container IP Address: 172.25.200.152",
                           "Container Hostname  : bcau",
                           "Container Dns Name  : bcau",
                           "Web Client          : http://bcau/BC/?tenant=default",
                           "Dev. Server         : http://bcau",
                           "Dev. ServerInstance : BC",
                           "Dev. Server Tenant  : default",
                           "Setting bcau to 172.25.200.152 in host hosts file",
                           "Setting bcau-default to 172.25.200.152 in host hosts file",
                           "",
                           "Files:",
                           "http://bcau:8080/ALLanguage.vsix",
                           "",
                           "WARNING: You are running a container which is 80 days old.",
                           "Microsoft recommends that you always run the latest version of our containers.",
                           "",
                           "Container Total Physical Memory is 15.8Gb",
                           "Container Free Physical Memory is 3.4Gb",
                           "",
                           "Initialization took 181 seconds",
                           "Ready for connections!",
                           "Initializing...",
                           "Setting host.docker.internal to 192.168.1.78 in container hosts file (copy from host hosts file)",
                           "Setting gateway.docker.internal to 192.168.1.78 in container hosts file (copy from host hosts file)",
                           "Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)",
                           "Setting host.containerhelper.internal to 172.25.192.1 in container hosts file",
                           "Restarting Container",
                           "PublicDnsName unchanged",
                           "Hostname is bcau",
                           "PublicDnsName is bcau",
                           "Using NavUserPassword Authentication",
                           "Starting Local SQL Server",
                           "Starting Internet Information Server",
                           "Starting Service Tier",
                           "Container IP Address: 172.25.199.32",
                           "Container Hostname  : bcau",
                           "Container Dns Name  : bcau",
                           "Web Client          : http://bcau/BC/?tenant=default",
                           "Dev. Server         : http://bcau",
                           "Dev. ServerInstance : BC",
                           "Dev. Server Tenant  : default",
                           "Setting bcau to 172.25.199.32 in host hosts file",
                           "Setting bcau-default to 172.25.199.32 in host hosts file",
                           "",
                           "Files:",
                           "http://bcau:8080/ALLanguage.vsix",
                           "",
                           "Container Total Physical Memory is 15.8Gb",
                           "Container Free Physical Memory is 2.4Gb",
                           "",
                           "Initialization took 15 seconds",
                           "Ready for connections!",
                           "Initializing...",
                           "Setting host.docker.internal to 10.3.32.68 in container hosts file (copy from host hosts file)",
                           "Setting gateway.docker.internal to 10.3.32.68 in container hosts file (copy from host hosts file)",
                           "Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)",
                           "Setting host.containerhelper.internal to 172.25.192.1 in container hosts file",
                           "Restarting Container",
                           "PublicDnsName unchanged",
                           "Hostname is bcau",
                           "PublicDnsName is bcau",
                           "Using NavUserPassword Authentication",
                           "Starting Local SQL Server",
                           "Starting Internet Information Server",
                           "Starting Service Tier",
                           "Container IP Address: 172.25.196.133",
                           "Container Hostname  : bcau",
                           "Container Dns Name  : bcau",
                           "Web Client          : http://bcau/BC/?tenant=default",
                           "Dev. Server         : http://bcau",
                           "Dev. ServerInstance : BC",
                           "Dev. Server Tenant  : default",
                           "Setting bcau to 172.25.196.133 in host hosts file",
                           "Setting bcau-default to 172.25.196.133 in host hosts file",
                           "",
                           "Files:",
                           "http://bcau:8080/ALLanguage.vsix",
                           "",
                           "Container Total Physical Memory is 15.8Gb",
                           "Container Free Physical Memory is 4.1Gb",
                           "",
                           "Initialization took 13 seconds",
                           "Ready for connections!",
                           "Initializing...",
                           "Setting host.docker.internal to 10.3.32.68 in container hosts file (copy from host hosts file)",
                           "Setting gateway.docker.internal to 10.3.32.68 in container hosts file (copy from host hosts file)",
                           "Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)",
                           "Setting host.containerhelper.internal to 172.25.192.1 in container hosts file",
                           "Restarting Container",
                           "Hostname is bcau",
                           "PublicDnsName is bcau",
                           "Using NavUserPassword Authentication",
                           "Starting Local SQL Server",
                           "Starting Internet Information Server",
                           "Creating Self Signed Certificate",
                           "Self Signed Certificate Thumbprint 8C2D44CCE704A0E488D69B1658A1ABCF33F3F199",
                           "Modifying Service Tier Config File with Instance Specific Settings",
                           "Starting Service Tier",
                           "Registering event sources",
                           "Creating DotNetCore Web Server Instance",
                           "Enabling Financials User Experience",
                           "Enabling rewrite rule: Don\u0027t rewrite system files",
                           "Enabling rewrite rule: Already have tenant specified",
                           "Enabling rewrite rule: Hostname (without port) to tenant",
                           "Dismounting Tenant",
                           "Mounting Tenant",
                           "Mounting Database for default on server localhost\\SQLEXPRESS",
                           "Sync\u0027ing Tenant",
                           "Tenant is Operational",
                           "Container IP Address: 172.25.200.194",
                           "Container Hostname  : bcau",
                           "Container Dns Name  : bcau",
                           "Web Client          : http://bcau/BC/?tenant=default",
                           "Dev. Server         : http://bcau",
                           "Dev. ServerInstance : BC",
                           "Dev. Server Tenant  : default",
                           "Setting bcau to 172.25.200.194 in host hosts file",
                           "Setting bcau-default to 172.25.200.194 in host hosts file",
                           "",
                           "Files:",
                           "http://bcau:8080/ALLanguage.vsix",
                           "",
                           "Container Total Physical Memory is 15.8Gb",
                           "Container Free Physical Memory is 3.3Gb",
                           "",
                           "Initialization took 243 seconds",
                           "Ready for connections!"
                       ],
    "docker.info":  [
                        "Client:",
                        " Debug Mode: false",
                        "",
                        "Server:",
                        " Containers: 4",
                        "  Running: 3",
                        "  Paused: 0",
                        "  Stopped: 1",
                        " Images: 42",
                        " Server Version: 19.03.12",
                        " Storage Driver: windowsfilter",
                        "  Windows: ",
                        " Logging Driver: json-file",
                        " Plugins:",
                        "  Volume: local",
                        "  Network: ics internal l2bridge l2tunnel nat null overlay private transparent",
                        "  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog",
                        " Swarm: inactive",
                        " Default Isolation: hyperv",
                        " Kernel Version: 10.0 19041 (19041.1.amd64fre.vb_release.191206-1406)",
                        " Operating System: Windows 10 Enterprise Version 2004 (OS Build 19041.388)",
                        " OSType: windows",
                        " Architecture: x86_64",
                        " CPUs: 8",
                        " Total Memory: 15.78GiB",
                        " Name: DESKTOP-S4PPI5V",
                        " ID: SBV3:PDQU:4ZEL:C73H:7EHO:DA5Q:MFTR:LDK5:AN6K:QKET:RT4F:U4SW",
                        " Docker Root Dir: C:\\ProgramData\\Docker",
                        " Debug Mode: true",
                        "  File Descriptors: -1",
                        "  Goroutines: 67",
                        "  System Time: 2020-10-08T16:01:38.3633132+13:00",
                        "  EventsListeners: 2",
                        " Registry: https://index.docker.io/v1/",
                        " Labels:",
                        " Experimental: false",
                        " Insecure Registries:",
                        "  127.0.0.0/8",
                        " Live Restore Enabled: false",
                        " Product License: Community Engine",
                        ""
                    ],
    "container.labels":  {
                             "country":  "AU",
                             "created":  "202010060338",
                             "cu":  "",
                             "eula":  "https://go.microsoft.com/fwlink/?linkid=861843",
                             "legal":  "http://go.microsoft.com/fwlink/?LinkId=837447",
                             "maintainer":  "Dynamics SMB",
                             "multitenant":  "Y",
                             "nav":  "",
                             "osversion":  "10.0.19041.388",
                             "platform":  "16.0.15884.17299",
                             "tag":  "0.1.0.23",
                             "version":  "16.5.15897.17318"
                         },
    "container.ping":  [
                           "",
                           "Pinging bcau [172.25.200.194] with 32 bytes of data:",
                           "Reply from 172.25.200.194: bytes=32 time\u003c1ms TTL=128",
                           "Reply from 172.25.200.194: bytes=32 time\u003c1ms TTL=128",
                           "Reply from 172.25.200.194: bytes=32 time\u003c1ms TTL=128",
                           "Reply from 172.25.200.194: bytes=32 time\u003c1ms TTL=128",
                           "",
                           "Ping statistics for 172.25.200.194:",
                           "    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),",
                           "Approximate round trip times in milli-seconds:",
                           "    Minimum = 0ms, Maximum = 0ms, Average = 0ms"
                       ],
    "container.env":  [
                          "accept_outdated=Y",
                          "ExitOnError=N",
                          "useSSL=N",
                          "accept_eula=Y",
                          "licenseFile=\u003cspecified\u003e",
                          "removePasswordKeyFile=Y",
                          "databaseInstance=",
                          "enableApiServices=Y",
                          "passwordKeyFile=c:\\run\\my\\aes.key",
                          "multitenant=Y",
                          "username=davidr",
                          "locale=en-AU",
                          "auth=NavUserPassword",
                          "databaseServer=",
                          "securePassword=\u003cspecified\u003e",
                          "DOTNET_RUNNING_IN_CONTAINER=true",
                          "COMPLUS_NGenProtectedProcess_FeatureEnabled=0",
                          "DatabaseName=CRONUS",
                          "IsBcSandbox=Y",
                          "artifactUrl=https://bcartifacts.azureedge.net/sandbox/16.5.15897.17318/au"
                      ]
}
```

**Additional context**
I tried to rule out BC by sending a mail from the docker PowerShell prompt. 

C:\WINDOWS\system32\CMD.EXE /C docker.exe exec -it bcau powershell -noexit c:\run\prompt.ps1
 
where bcau is my container name.
 
And from inside that PowerShell window I try the following snazzy script:

```
[string]$userName = 'david.roys@intergen.co.nz'
[string]$userPassword = 'ThisIsNotMyRealPassword'
[securestring]$secStringPassword = ConvertTo-SecureString $userPassword -AsPlainText -Force
[pscredential]$credObject = New-Object System.Management.Automation.PSCredential ($userName, $secStringPassword)
Send-MailMessage -From $userName -To $userName -Subject 'Test from PowerShell' -Body 'Body from Docker' -SmtpServer smtp.office365.com -Credential $credObject -UseSsl -Port 587
```
This returns:
 
Send-MailMessage : The remote name could not be resolved: 'smtp.office365.com'
At line:1 char:1
+ Send-MailMessage -From $userName -To $userName -Subject 'Test from PS ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.Mail.SmtpClient:SmtpClient) [Send-MailMessage], SmtpException
    + FullyQualifiedErrorId : SmtpException,Microsoft.PowerShell.Commands.SendMailMessage
 
Which is very similar to the message I get back from BC. The same script executed from a normal PowerShell prompt (i.e. not in Docker) works.

**Comment by freddydk** [comment]
Can you do 

ping www.google.com 

inside the container - does that resolve?

**Comment by DavidRoys** [comment]
Hi Freddy, I'm not sure what to report now. I'll just tell you exactly what happened without trying to interpret it to see if it makes any sense at all to you, but in short, it is now working.

I started my laptop when I returned home from the office.
I started docker desktop and it failed to start with an error about not being able to write to unprotected drives.
I ran my registry edit to allow docker to start properly (normally I run this first but I forgot this time).
I started docker desktop and it started OK.
My bcau container had a status of exited and I could not restart it.
I rebooted my laptop.
I noticed that my wifi was dead (seems to be a weird problem with my laptop where the wifi intermittently just dissappears).
I ran the repair on my wifi and connected to the internet.
I ran my regedit to allow writing to unprotected drives.
I started docker desktop.
All containers were started and healthy.
I tried the send test email from my bcau container. - IT WORKED!
I don't know why.
Just for completeness I started a PowerShell prompt in the container and tried to ping www.google.com - host was not found.
I started a CMD prompt in the container and tried to ping www.google.com - it resolved OK.
I ran a PowerShell prompt locally and tried ping www.google.com - host not found.
I tried CMD prompt locally and tried ping www.google.com - it resolved OK.

Like I said, I cannot make sense of any of this but it looks like it's more to do with my environment or equipment and it is currently working, so I'm happy for you to close this issue.
